pyomo/contrib/iis/mis.py

```diff
@@ -0,0 +1,371 @@
+        if pyo.check_optimal_termination(results):
+            logger.info(f"A feasible solution was found!")
+        else:
+            logger.info(
+                f"Could not find a feasible solution with violated constraints or bounds. This model is likely unstable"
+            )
+
+    # Phase 2 -- deletion filter
+    # remove slacks by fixing them to 0
+    for v in slack_block.component_data_objects(pyo.Var):
+        v.fix(0)
+    for o in modified_model.component_data_objects(pyo.Objective, descend_into=True):
+        o.deactivate()
+
+    # mark all constraints not in the filter as inactive
+    for c in modified_model.component_data_objects(pyo.Constraint):
+        if c in elastic_filter:
+            continue
+        else:
+            c.deactivate()
+
+    try:
+        results = solver.solve(modified_model, tee=tee)
+    except:
+        results = None
+
+    if pyo.check_optimal_termination(results):
+        msg += "Could not determine Minimal Intractable System\n"
+    else:
+        deletion_filter = []
+        guards = []
+        for constr in elastic_filter:
+            constr.deactivate()
+            for var, val in _modified_model_value_cache.items():
+                var.set_value(val, skip_validation=True)
+            math_failure = False
+            try:
+                results = solver.solve(modified_model, tee=tee)
+            except:
+                math_failure = True
+
+            if math_failure:
+                constr.activate()
+                guards.append(constr)
+            elif pyo.check_optimal_termination(results):
+                constr.activate()
+                deletion_filter.append(constr)
+            else:  # still infeasible without this constraint
+                pass
+
+        msg += "Computed Minimal Intractable System (MIS)!\n"
+        msg += "Constraints / bounds in MIS:\n"
+        msg = _get_results(deletion_filter, msg)
+        msg += "Constraints / bounds in guards for stability:"
+        msg = _get_results(guards, msg)
+
+    logger.info(msg)
+
+
+def _get_results_with_value(constr_value_generator, msg=None):
+    if msg is None:
+        msg = ""
+    for c, value in constr_value_generator:
+        c_name = c.name
+        if "_variable_bounds" in c_name:
+            name = c.local_name
+            if "lb" in name:
+                msg += f"\tlb of var {name[7:]} by {value}\n"
+            elif "ub" in name:
+                msg += f"\tub of var {name[7:]} by {value}\n"
+            else:
+                raise RuntimeError("unrecognized var name")
+        else:
+            msg += f"\tconstraint: {c_name} by {value}\n"
+    return msg
+
+
+def _get_results(constr_generator, msg=None):
+    if msg is None:
+        msg = ""
+    for c in constr_generator:
+        c_name = c.name
+        if "_variable_bounds" in c_name:
+            name = c.local_name
+            if "lb" in name:
+                msg += f"\tlb of var {name[7:]}\n"
```

Comment:
7 is a magic number here. I assume that you ignore the first 7 chars because they are random cruft, but can you either pull this out to its own variable (e.g., `ignore_chars = 7`) or put a comment to say why you are ignoring them?

(Because this is the same as above, might be worth putting it as a global var)